### PR TITLE
Remove sourceMaps on web

### PR DESCRIPTION
**Pull request by bartlomiejbloniarz**
## Summary

This PR removes sourceMaps on web, significantly reducing the bundle size. The main change is introduced in `util.ts`. Remaining changes were introduced for bundle size testing purposes.

## Test plan
Run `yarn build` in `NextExample`. The difference between `reanimated.js` and `noreanimated.js` should be around 50KB.

### CODE REVIEW

File: NextExample/next.config.js [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+module.exports = withPlugins([withBundleAnalyzer, withExpo], {
   // transpilePackages is a Next.js +13.1 feature.
   // older versions can use next-transpile-modules
   eslint: {
     ignoreDuringBuilds: true,
   },
   transpilePackages: ['react-native-reanimated', 'react-native', 'expo'],
   webpack(config, _options) {
+    // uncomment this to remove bundle minification
+    // config.optimization.minimizer = [];
```
Comment (tomekzaw) [inline NextExample/next.config.js:19]: Can we enable this conditionally based on some constant or environmental variable?

File: NextExample/pages/noreanimated.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { StyleSheet, View, Text } from 'react-native';
+
+export default function Reanimated() {
+  return (
+    <View style={styles.container}>
+      <Text sha accessibilityRole="header" style={styles.text}>
+        Reanimated Page
+      </Text>
+      <Text style={styles.link} accessibilityRole="link" href={`/`}>
```
Comment (tomekzaw) [inline NextExample/pages/noreanimated.js:9]:
```suggestion
      <Text style={styles.link} accessibilityRole="link" href="/">
```

File: NextExample/pages/reanimated.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { StyleSheet } from 'react-native';
+import Animated from 'react-native-reanimated';
+
+export default function Reanimated() {
+  return (
+    <Animated.View style={styles.container}>
+      <Animated.Text sha accessibilityRole="header" style={styles.text}>
+        Reanimated Page
+      </Animated.Text>
+      <Animated.Text style={styles.link} accessibilityRole="link" href={`/`}>
```
Comment (tomekzaw) [inline NextExample/pages/reanimated.js:10]:
```suggestion
      <Animated.Text style={styles.link} accessibilityRole="link" href="/">
```

File: NextExample/next.config.js [hunk 4]
```diff
@@ -1,14 +1,22 @@
 const { withExpo } = require('@expo/next-adapter');
 
+const withPlugins = require('next-compose-plugins');
+
+const withBundleAnalyzer = require('@next/bundle-analyzer')({
+  enabled: process.env.ANALYZE === 'true',
```
Comment (tomekzaw) [inline NextExample/next.config.js:6]: Usually environmental variables are enabled with `1`, what do you think?

**Review by tomekzaw — APPROVED** [review]
LGTM, left some minor suggestions.

File: NextExample/next.config.js [hunk 5]
```diff
@@ -1,14 +1,25 @@
 const { withExpo } = require('@expo/next-adapter');
 
+const withPlugins = require('next-compose-plugins');
+
+const withBundleAnalyzer = require('@next/bundle-analyzer')({
+  enabled: process.env.ANALYZE === '1',
```
Comment (tomekzaw) [inline NextExample/next.config.js:6]: I'm afraid `ANALYZE` is not informative enough, can we rename it to `ANALYZE_BUNDLE` or `ENABLE_BUNDLE_ANALYZER`?

File: NextExample/next.config.js [hunk 6]
```diff
@@ -1,14 +1,25 @@
...
+});
+
+const disableMinification = process.env.DISABLE_MINIFICATION === '1';
```
Comment (tomekzaw) [inline NextExample/next.config.js:9]: Can we add a short comment here about the effect of this flag?

File: NextExample/next.config.js [hunk 7]
```diff
@@ -1,14 +1,25 @@
...
   // transpilePackages is a Next.js +13.1 feature.
   // older versions can use next-transpile-modules
   eslint: {
     ignoreDuringBuilds: true,
   },
   transpilePackages: ['react-native-reanimated', 'react-native', 'expo'],
   webpack(config, _options) {
+    if (disableMinification) {
+      config.optimization.minimizer = [];
+    }
```
Comment (piaskowyk) [inline NextExample/next.config.js:22]: I think we should add to documentation information what does mean DISABLE_MINIFICATION and ANALYZE, and how to use it.

**Review by piaskowyk — CHANGES_REQUESTED** [review]

File: plugin/src/utils.ts [hunk 8]
```diff
@@ -1,3 +1,6 @@
 export function isRelease() {
-  return !!process.env.BABEL_ENV?.match(/(prod|release|stag[ei])/i);
+  const regex = /(prod|release|stag[ei])/i;
+  return !!(
+    process.env.BABEL_ENV?.match(regex) || process.env.NODE_ENV?.match(regex)
```
Comment (tjzel) [inline plugin/src/utils.ts:4]:
```suggestion
  const pattern = /(prod|release|stag[ei])/i;
  return !!(
    process.env.BABEL_ENV?.match(pattern) || process.env.NODE_ENV?.match(pattern)
```

File: NextExample/pages/noreanimated.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import { StyleSheet, View, Text } from 'react-native';
+
+export default function Reanimated() {
```
Comment (tjzel) [inline NextExample/pages/noreanimated.js:3]: Or `App` or something else, `Reanimated` is weird here.
```suggestion
export default function Page() {
```

**Review by tjzel — CHANGES_REQUESTED** [review]

**Review by tjzel — APPROVED** [review]

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**